sbin/spark-config.sh

```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env bash
```

Comment:
Is this a safe change, @stczwd ? This looks like that we need to handle in a separate PR, not in this SPARK-38433, to give a proper visibility.

Reply:
This chage is because [SC2148](https://github.com/koalaman/shellcheck/wiki/SC2148).

Reply:
> This chage is because [SC2148](https://github.com/koalaman/shellcheck/wiki/SC2148).

@stczwd . I'm asking about the behavior change of `bash` binary in Spark environment. Your change could switch the `bash` location from `/bin/bash` to `/usr/bin/bash` silently. Frequently, those two bash versions are not the same.

Reply:
Hm, actually, all the other scripts under sbin directory use `#!/usr/bin/env bash`.